sdk/tables/azure-data-tables/src/table_clients.cpp

```diff
@@ -469,14 +470,11 @@ Azure::Response<Models::UpdateEntityResult> TableClient::UpdateEntity(
   request.SetHeader(ContentLengthHeader, std::to_string(requestBody.Length()));
   request.SetHeader(AcceptHeader, AcceptFullMeta);
   request.SetHeader(PreferHeader, PreferNoContent);
-  if (!tableEntity.GetETag().Value.empty())
+
+  if (isUpsert == false && !tableEntity.GetETag().Value.empty())
```

Comment:
Do you need to check `GetETag().HasValue()` before calling `GetETag().Value()`, or will the etag always have a value at this point?